src/software/thunderscope/wifi_communication_manager.py

```diff
@@ -201,7 +201,10 @@ def __receive_robot_status(self, robot_status: Message) -> None:
         )
         self.__forward_to_proto_unix_io(
             RobotStatistic,
-            RobotStatistic(round_trip_time_seconds=round_trip_time_seconds),
+            RobotStatistic(
+                robot_id=robot_status.robot_id,
+                round_trip_time_seconds=round_trip_time_seconds,
+            ),
```

Comment:
This is the actual part of the PR that fixes the bug causing `RobotInfo` to display no battery status. The `RobotStatistic` wasn't being constructed with a `robot_id`, so it would default to an ID of 0 and hence only the `RobotInfo` for robot 0 would get `RobotStatistic` updates (`RobotInfo` needs to receive `RobotStatistic` in order to update its UI)